### PR TITLE
ed: Add admin to stop points and poi using cities

### PR DESCRIPTION
**Pull request by TeXitoi**

### CODE REVIEW

File: source/type/data.cpp [hunk 1]
```diff
@@ -167,8 +171,35 @@ void Data::build_proximity_list(){
...
+    for (georef::POI* poi: geo_ref->pois) {
+        if (poi->coord.is_initialized()) {
+            const auto &admins = find_admins(poi->coord);
+            boost::push_back(poi->admin_list, admins);
+            if (admins.empty()) ++cpt_no_projected;
+        } else {
+            cpt_no_initialized++;
+        }
+    }
+    LOG4CPLUS_DEBUG(log, cpt_no_projected << "/" << geo_ref->pois.size()
```
Comment (antoine-de) [inline source/type/data.cpp]: I think you should wrap those with `if(cpt_no_projected)` for log clarity

File: source/type/data.cpp [hunk 2]
```diff
@@ -167,8 +171,35 @@ void Data::build_proximity_list(){
...
+    auto log = log4cplus::Logger::getInstance("ed::Data");
+
+    // set admins to stop points
+    int cpt_no_projected = 0;
+    for(type::StopPoint* stop_point : pt_data->stop_points) {
+        const auto &admins = find_admins(stop_point->coord);
+        boost::push_back(stop_point->admin_list, admins);
+        if (admins.empty()) ++cpt_no_projected;
+    }
+    LOG4CPLUS_DEBUG(log, cpt_no_projected << "/" << pt_data->stop_points.size()
```
Comment (antoine-de) [inline source/type/data.cpp]: wrap that too

File: source/ed/ed2nav.cpp [hunk 3]
```diff
@@ -103,6 +107,45 @@ int main(int argc, char * argv[])
     now = start = pt::microsec_clock::local_time();
 
     ed::EdReader reader(connection_string);
+
+    std::unique_ptr<pqxx::connection> find_admin_connection;
```
Comment (antoine-de) [inline source/ed/ed2nav.cpp]: can you put that in a separate function to keep the main smaller ?

And It think some comments would be nice

File: source/ed/ed2nav.cpp [hunk 4]
```diff
@@ -103,6 +107,45 @@ int main(int argc, char * argv[])
...
+            std::stringstream request;
+            request << "SELECT uri, name, insee, level, post_code, "
+                    << "ST_X(coord::geometry) as lon, ST_Y(coord::geometry) as lat "
+                    << "FROM administrative_regions "
+                    << "WHERE ST_DWithin(ST_GeographyFromText('POINT("
+                    << std::setprecision(16) << c.lon() << " " << c.lat() << ")'), boundary, 0)";
+            pqxx::work work(*find_admin_connection);
+            pqxx::result result = work.exec(request);
+            std::vector<navitia::georef::Admin*> res;
+            for (auto it = result.begin(); it != result.end(); ++it) {
```
Comment (antoine-de) [inline source/ed/ed2nav.cpp]: Is it possible to have 2 admins ?  I  don't think so in the normal case, shouldn't we log those case ?

File: source/ed/ed2nav.cpp [hunk 5]
```diff
@@ -103,6 +107,45 @@ int main(int argc, char * argv[])
...
+                    admin = data.geo_ref->admins.back();
+                    admin->comment = "from cities";
+                    it["uri"].to(admin->uri);
+                    it["name"].to(admin->name);
+                    it["insee"].to(admin->insee);
+                    it["level"].to(admin->level);
+                    it["post_code"].to(admin->post_code);
+                    admin->coord.set_lon(it["lon"].as<double>());
+                    admin->coord.set_lat(it["lat"].as<double>());
+                }
```
Comment (antoine-de) [inline source/ed/ed2nav.cpp]: I think it would be nice to log the number of not found admin

File: source/ed/ed2nav.cpp [hunk 6]
```diff
@@ -103,6 +107,45 @@ int main(int argc, char * argv[])
...
+                    data.geo_ref->admins.push_back(new navitia::georef::Admin());
+                    admin = data.geo_ref->admins.back();
+                    admin->comment = "from cities";
+                    it["uri"].to(admin->uri);
+                    it["name"].to(admin->name);
+                    it["insee"].to(admin->insee);
+                    it["level"].to(admin->level);
+                    it["post_code"].to(admin->post_code);
+                    admin->coord.set_lon(it["lon"].as<double>());
+                    admin->coord.set_lat(it["lat"].as<double>());
```
Comment (antoine-de) [inline source/ed/ed2nav.cpp]: don't you also need the boundary ?

Comment (TeXitoi) [inline source/ed/ed2nav.cpp]: There is not boundary in geopal

**Comment by TeXitoi** [comment]
Jira: http://jira.canaltp.fr/browse/NAVITIAII-1140

**Pull request merged**